rasp/base.py

```diff
@@ -67,9 +67,12 @@ def as_func(cls, method_name):
         try:
             return getattr(TmpEngine, method_name)
         except AttributeError:
-            raise NotImplementedError('Class {} does not implement {}'.format(class_name, func_name))
+            raise NotImplementedError('Class {} does not implement {}'.format(class_name, method_name))
 
-    def get_page_source(self, url, params=None, headers=None):
+    def get_page_source(self, url,
+                        params=None,
+                        headers=None,
+                        pre_fetch_callback=None):
```

Comment:
I think it should be.  Ideally, the `get_page_source` doesn't handle anything engine specific, or else the crawlers also have to be aware of what kind of engine they are using, which is sub-optimal.